### PR TITLE
add pam auth file for 2204

### PR DESCRIPTION
**Pull request by anujmaheshwari1**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: vhdbuilder/packer/vhd-image-builder-arm64-gen2.json [hunk 1]
```diff
@@ -190,6 +190,11 @@
       "source": "parts/linux/cloud-init/artifacts/pam-d-common-auth",
       "destination": "/home/packer/pam-d-common-auth"
     },
+    {
+      "type": "file",
+      "source": "parts/linux/cloud-init/artifacts/pam-d-common-auth-2204",
+      "destination": "/home/packer/pam-d-common-auth-2204"
```
Comment (alexeldeib) [inline vhdbuilder/packer/vhd-image-builder-arm64-gen2.json:196]:
```suggestion
      "destination": "/home/packer/pam-d-common-auth"
```

File: vhdbuilder/packer/vhd-image-builder-base.json [hunk 2]
```diff
@@ -202,6 +202,11 @@
       "source": "parts/linux/cloud-init/artifacts/pam-d-common-auth",
       "destination": "/home/packer/pam-d-common-auth"
     },
+    {
+      "type": "file",
+      "source": "parts/linux/cloud-init/artifacts/pam-d-common-auth-2204",
+      "destination": "/home/packer/pam-d-common-auth-2204"
```
Comment (alexeldeib) [inline vhdbuilder/packer/vhd-image-builder-base.json:208]:
```suggestion
      "destination": "/home/packer/pam-d-common-auth"
```

File: vhdbuilder/packer/packer_source.sh [hunk 3]
```diff
@@ -100,6 +100,10 @@ copyPackerFiles() {
     cpAndMode $AKS_LOGROTATE_TIMER_DROPIN_SRC $AKS_LOGROTATE_TIMER_DROPIN_DEST 644
   fi
 
+  if [[ ${UBUNTU_RELEASE} == "22.04" ]]; then
+    PAM_D_COMMON_AUTH_SRC=/home/packer/pam-d-common-auth-2204
+  fi
+
```
Comment (alexeldeib) [inline vhdbuilder/packer/packer_source.sh:106]:
```
```

File: vhdbuilder/packer/packer_source.sh [hunk 4]
```diff
@@ -100,6 +100,10 @@ copyPackerFiles() {
     cpAndMode $AKS_LOGROTATE_TIMER_DROPIN_SRC $AKS_LOGROTATE_TIMER_DROPIN_DEST 644
   fi
 
+  if [[ ${UBUNTU_RELEASE} == "22.04" ]]; then
```
Comment (alexeldeib) [inline vhdbuilder/packer/packer_source.sh:103]: delete this and do the thing below ;)

File: parts/linux/cloud-init/artifacts/pam-d-common-auth-2204 [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+auth	requisite			pam_deny.so
+# prime the stack with a positive return value if there isn't one already;
+# this avoids us returning an error just because nothing sets a success code
+# since the modules above will each just jump around
+auth	required			pam_permit.so
+# and here are more per-package modules (the "Additional" block)
+# end of pam-auth-update config
+
+# 5.3.2 Ensure lockout for failed password attempts is configured
+auth required pam_faillock.so preauth silent audit deny=5 unlock_time=900
```
Comment (alexeldeib) [inline parts/linux/cloud-init/artifacts/pam-d-common-auth-2204:28]: we could just remove this one line from both files, and add a new drop-in too -- we already have a password file too

**Comment by alexeldeib** [comment]
need a `make generate` I think

**Review by chengliangli0918 — APPROVED** [review]

File: vhdbuilder/packer/packer_source.sh (hunk 4 above)
Comment (alexeldeib) [reply vhdbuilder/packer/packer_source.sh:103]: nevermind -- synced offline. my brain was broken :D

**Pull request merged**